app/models/service_template.rb

```diff
@@ -64,17 +64,27 @@ class ServiceTemplate < ApplicationRecord
   virtual_column   :template_valid_error_message, :type => :string
   virtual_column   :archived,                     :type => :boolean
   virtual_column   :active,                       :type => :boolean
+  virtual_column   :schedule_time,                :type => :time
 
   default_value_for :service_type, 'unknown'
   default_value_for(:generic_subtype) { |st| 'custom' if st.prov_type == 'generic' }
 
   virtual_has_one :config_info, :class_name => "Hash"
+  virtual_has_one :miq_schedule,:class_name => "MiqSchedule"
 
   scope :with_service_template_catalog_id,          ->(cat_id) { where(:service_template_catalog_id => cat_id) }
   scope :without_service_template_catalog_id,       ->         { where(:service_template_catalog_id => nil) }
   scope :with_existent_service_template_catalog_id, ->         { where.not(:service_template_catalog_id => nil) }
   scope :displayed,                                 ->         { where(:display => true) }
 
+  def schedule_time
+    miq_schedule.try(:run_at).try(:[], :start_time)
+  end
+
+  def miq_schedule
+    MiqSchedule.where(:towhat => "ServiceTemplate").find { |s| s.resource_id == id }
```

Comment:
Isn't this exactly what Brandon was trying to avoid with his association?  This can just be has_one :miq_schedule, no?

Reply:
Can I do that with the `reserves` table containing the foreign key? I thought I had to do it this way. The plan is that we remove the `reserves` once this is backported and then we make it a normal association.

Reply:
yeah, in the future I think.  We want to use reserves for now, backport, then fix this to not use reserves and get it to be  a real association.

Reply:
That was my plan, but shouldn't this be `miq_schedules` (plural)?

Reply:
I was expecting that a `ServiceTemplate` would have one schedule